Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas

```diff
@@ -496,21 +510,25 @@ procedure TFormAdjustDesignators.ButtonOKClick(Sender: TObject);
             if Size <  MinimumHeight then
                 Size := MinimumHeight;
 
+            Designator.Size := Size;
 
-            // Set the Designator AutoPosition to the center-center but stop comment being moved.
-            Designator.SetState_MultilineTextAutoPosition(eAutoPos_CenterLeft);
-            Designator.SetState_Multiline(false);
+            Designator.SetState_Multiline(true);
+            Designator.MultilineTextHeight := 0;
+            Designator.MultilineTextWidth  := 0;
+            Designator.SetState_Multiline(true);
+            Designator.SetState_MultilineTextAutoPosition(eAutoPos_CenterCenter);   // CenterLeft
+//            Designator.UpdateTextPosition;
 
-            Designator.Size := Size;
             Designator.SetState_XSizeYSize;
             Designator.GraphicallyInvalidate;
             Component.Name.EndModify;
 
+//     Set the Designator AutoPosition to the center-center but stop comment being moved.
             Component.ChangeCommentAutoposition( eAutoPos_Manual );
             Component.ChangeNameAutoposition(eAutoPos_CenterCenter);
             Component.SetState_NameAutoPos(eAutoPos_CenterCenter);

```

Comment:
Does following `ChangeNameAutoposition(eAutoPos_CenterCenter)` with `SetState_NameAutoPos(eAutoPos_CenterCenter)` do something particular? Does it avoid having to do `EndModify` immediately followed by `BeginModify` to get object property cache to update for later manipulation?

Reply:
Yes there is a reason. This script uses the hacky method of reusing (abusing) the real designator. The autoposition functions of designator & comment interfere with each other so I use a property shuffle trick to stop Altium from moving anything.

Reply:
Yeah, I understand that part, where you set the comment to manual autoposition so it doesn't move after the designator "gets out of the way", what I'm not clear on is the double-setting of the designator autoposition using both `ChangeNameAutoposition` function and the `SetState_NameAutoPos` procedure, one right after the other. It looks redundant, but I'm wondering if it actually has a useful effect.

In my scripting I've noticed that certain changes to autoposition don't really "take effect" until I call EndModify on the text, and `ChangeNameAutoposition` seems to work more immediately. I was just wondering if you had some insight or experience on the nuance behind these calls that I'm still trying to grok.

Reply:
Yeah, I think that extra SetState_NameAutoPos is redundant. It does not trigger the action like ChangeName & AutoPosition_NameComment do.
SetState_NameAutoPos  (fortunately) allows you to set the original auto setting back without damage.
That block of code was copied from my original designator script that uses Comment as a 2nd designator.